src/content/docs/en/reference/programmatic-reference.mdx

```diff
@@ -0,0 +1,197 @@
+  root: "./my-project",
+});
+
+// Stop the server if needed
+await devServer.stop();
+```
+
+### `DevServer`
+
+```ts
+export interface DevServer {
+	address: AddressInfo;
+	handle: (req: http.IncomingMessage, res: http.ServerResponse<http.IncomingMessage>) => void;
+	watcher: vite.FSWatcher;
+	stop(): Promise<void>;
+}
+```
+
+#### `address`
+
+The address the dev server is listening on.
+
+This property contains the value returned by Node's [`net.Server#address()` method](https://nodejs.org/api/net.html#serveraddress).
+
+#### `handle()`
+
+A handle for raw Node HTTP requests. You can call `handle()` with an [`http.IncomingMessage`](https://nodejs.org/api/http.html#class-httpincomingmessage) and an [`http.ServerResponse`](https://nodejs.org/api/http.html#class-httpserverresponse) instead of sending a request through the network.
+
+#### `watcher`
+
+The [Chokidar file watcher](https://github.com/paulmillr/chokidar#getting-started) as exposed by [Vite's development server](https://vite.dev/guide/api-javascript#vitedevserver).
+
+#### `stop()`
+
+Stops the development server. This closes all idle connections and stops listening for new connections.
+
+Returns a `Promise` that resolves once all pending requests have been fulfilled and all idle connections have been closed.
+
+## `build()`
+
+**Type:** `(inlineConfig: AstroInlineConfig) => Promise<void>`
+
+Similar to [`astro build`](#astro-build), it builds your site for deployment.
+
+```js
+import { build } from "astro";
+
+await build({
+  root: "./my-project",
+});
+```
+
+## `preview()`
+
+**Type:** `(inlineConfig: AstroInlineConfig) => Promise<PreviewServer>`
+
+Similar to [`astro preview`](#astro-preview), it starts a local server to serve your build output.
+
+If no adapter is set in the configuration, the preview server will only serve the built static files.
+If an adapter is set in the configuration, the preview server is provided by the adapter. Adapters are not required to provide a preview server, so this feature may not be available depending on your adapter of choice.
+
+```js
+import { preview } from "astro";
+
+const previewServer = await preview({
+  root: "./my-project",
+});
+
+// Stop the server if needed
+await previewServer.stop();
+```
+
+### `PreviewServer`
+
+```ts
+export interface PreviewServer {
+	host?: string;
+	port: number;
+	closed(): Promise<void>;
+	stop(): Promise<void>;
+}
+```
+
+#### `host`
+
+The host where the server is listening for connections.
+
+Adapters are allowed to leave this field unset. The value of `host` is implementation-specific.
+
+#### `port`
+
+The port where the server is listening for connections.
+
+#### `stop()`
+
+Asks the preview server to close, stop accepting requests, and drop idle connections.
+
+The returned `Promise` resolves when the close request has been sent. This does not mean that the server has closed yet. Use the [`closed()`](#closed) method if you need to ensure the server has fully closed.
+
+#### `closed()`
```

Comment:
Same remark as before regarding the format to use... `() => ...` or directly the output.
```suggestion
#### `closed()`

<p>

**Type:** `Promise<void>`
</p>
```